### PR TITLE
Hive SSO guides for Okta and Azure AD

### PR DESCRIPTION
**Pull request by YassinEldeeb**
### Background

Documents all of the major identity providers services connectivity with Hive's SSO

### Description

- [x] Okta
- [x] Azure AD

### CODE REVIEW

**Review by dotansimha — APPROVED** [review]

File: packages/web/docs/src/pages/features/sso-oidc-provider.mdx [hunk 1]
```diff
@@ -55,3 +55,180 @@ Provider. Share this link with your users or add it to your SSO provider dashboa
 people will be redirected to it.
 
 E.g. `https://app.graphql-hive.com/auth/oidc?id=xxxxxxxx-1234-1234-1234-xxxxxxxxxxxx`
+
+## Configuring with Okta
+
+First we need to have an Okta account, so you can register for an account from
```
Comment (kamilkisiela) [inline packages/web/docs/src/pages/features/sso-oidc-provider.mdx:61]: I think we can skip the trial part. If somebody is interested in connecting Okta to Hive, they probably have Okta account already.

File: packages/web/docs/src/pages/features/sso-oidc-provider.mdx [hunk 2]
```diff
@@ -55,3 +55,180 @@ Provider. Share this link with your users or add it to your SSO provider dashboa
...
+https://login.okta.com/signin/register or sign up for a 30 days trial from
+https://www.okta.com/free-trial to ease the process of going through this guide.
+
+> Note: that unfortunately for sigining up for an Okta account, you're required to have a busieness
+> email.
```
Comment (kamilkisiela) [inline packages/web/docs/src/pages/features/sso-oidc-provider.mdx:66]: It's the requirement of Okta, we should skip it in our docs.

File: packages/web/docs/src/pages/features/sso-oidc-provider.mdx [hunk 3]
```diff
@@ -55,3 +55,180 @@ Provider. Share this link with your users or add it to your SSO provider dashboa
...
+You can use the link provided in the modal right section to let your users login via Okta.
+
+![](/docs/pages/features/sso-oidc-provider/hive-manage-openid-okta-login-url.png)
+
+Hope that helped, and let us know if something changed on either Okta's or Hive's side in the
+screenshots and/or steps and we'll make that change happen.
+
+## Configuring with Azure AD
+
+First we need to have a Microsoft Azure account, so you can register for a free account from
```
Comment (kamilkisiela) [inline packages/web/docs/src/pages/features/sso-oidc-provider.mdx:131]: same as with Okta

**Review by kamilkisiela — CHANGES_REQUESTED** [review]
Overall looks, good. I added few comments.

Do you think it makes sense to use a tool like https://squoosh.app/ or something else to reduce the size of the images?

**Comment by dotansimha** [comment]
> Do you think it makes sense to use a tool like [squoosh.app](https://squoosh.app/) or something else to reduce the size of the images?

I think we have image optimization with next?

**Comment by kamilkisiela** [comment]
@dotansimha you might be right

File: packages/web/docs/src/pages/features/sso-oidc-provider.mdx (hunk 1 above)
Comment (YassinEldeeb) [reply packages/web/docs/src/pages/features/sso-oidc-provider.mdx:61]: Yeah, agree

File: packages/web/docs/src/pages/features/sso-oidc-provider.mdx (hunk 2 above)
Comment (YassinEldeeb) [reply packages/web/docs/src/pages/features/sso-oidc-provider.mdx:66]: ok

File: packages/web/docs/src/pages/features/sso-oidc-provider.mdx [hunk 4]
```diff
@@ -55,3 +55,180 @@ Provider. Share this link with your users or add it to your SSO provider dashboa
...
+
+## Configuring with Okta
+
+First we need to have an Okta account, so you can register for an account from
+https://login.okta.com/signin/register or sign up for a 30 days trial from
+https://www.okta.com/free-trial to ease the process of going through this guide.
+
+> Note: that unfortunately for sigining up for an Okta account, you're required to have a busieness
+> email.
+
```
Comment (YassinEldeeb) [inline packages/web/docs/src/pages/features/sso-oidc-provider.mdx:67]:
```suggestion
```

File: packages/web/docs/src/pages/features/sso-oidc-provider.mdx [hunk 5]
```diff
@@ -55,3 +55,180 @@ Provider. Share this link with your users or add it to your SSO provider dashboa
...
+![](/docs/pages/features/sso-oidc-provider/hive-manage-openid-okta-login-url.png)
+
+Hope that helped, and let us know if something changed on either Okta's or Hive's side in the
+screenshots and/or steps and we'll make that change happen.
+
+## Configuring with Azure AD
+
+First we need to have a Microsoft Azure account, so you can register for a free account from
+https://azure.microsoft.com/en-us/free
```
Comment (YassinEldeeb) [inline packages/web/docs/src/pages/features/sso-oidc-provider.mdx:132]:
```suggestion
```

File: packages/web/docs/src/pages/features/sso-oidc-provider.mdx [hunk 6]
```diff
@@ -55,3 +55,180 @@ Provider. Share this link with your users or add it to your SSO provider dashboa
...
+  so that they're able to login. Azure made it extra annoying because the only way to add an email
+  is by editing an existing user not when creating a user record, but it is what it is. You can
+  follow the steps in the screenshots below to be able to add an email for a user record.
+  ![](/docs/pages/features/sso-oidc-provider/azure-ad-users-tab.png)
+  ![](/docs/pages/features/sso-oidc-provider/azure-ad-users-table.png)
+  ![](/docs/pages/features/sso-oidc-provider/azure-ad-edit-user-properties.png)
+  ![](/docs/pages/features/sso-oidc-provider/azure-ad-edit-user-email-field.png)
+</Callout>
+
+And...Woohoo, you now have Azure AD connected! 🥳🎊
```
Comment (YassinEldeeb) [inline packages/web/docs/src/pages/features/sso-oidc-provider.mdx:230]:
```suggestion
And congrats, you know have Azure AD connected! 🥳🎉
```

File: packages/web/docs/src/pages/features/sso-oidc-provider.mdx [hunk 7]
```diff
@@ -55,3 +55,171 @@ Provider. Share this link with your users or add it to your SSO provider dashboa
...
+clicking the copy button next to them, then go back to your Okta's app SSO configuration and scroll
+down to the "General Settings" section and click "Edit" on the right, then scroll down to the
+"LOGIN" section and paste both the Sign-in redirect URI and the Sign-out redirect URI from Hive into
+their responding inputs and click "Save" when you're done.
+
+![](/docs/pages/features/sso-oidc-provider/hive-manage-openid-connect-modal.png)
+
+![](/docs/pages/features/sso-oidc-provider/okta-app-general-settings-section.png)
+
+And congrats, you know have Okta connected! 🥳🎉
```
Comment (YassinEldeeb) [inline packages/web/docs/src/pages/features/sso-oidc-provider.mdx:113]:
```suggestion
And congrats, you now have Okta connected! 🥳🎉
```

File: packages/web/docs/src/pages/features/sso-oidc-provider.mdx [hunk 8]
```diff
@@ -55,3 +55,171 @@ Provider. Share this link with your users or add it to your SSO provider dashboa
...
+  so that they're able to login. Azure made it extra annoying because the only way to add an email
+  is by editing an existing user not when creating a user record, but it is what it is. You can
+  follow the steps in the screenshots below to be able to add an email for a user record.
+  ![](/docs/pages/features/sso-oidc-provider/azure-ad-users-tab.png)
+  ![](/docs/pages/features/sso-oidc-provider/azure-ad-users-table.png)
+  ![](/docs/pages/features/sso-oidc-provider/azure-ad-edit-user-properties.png)
+  ![](/docs/pages/features/sso-oidc-provider/azure-ad-edit-user-email-field.png)
+</Callout>
+
+And congrats, you know have Azure AD connected! 🥳🎉
```
Comment (YassinEldeeb) [inline packages/web/docs/src/pages/features/sso-oidc-provider.mdx:221]:
```suggestion
And congrats, you now have Azure AD connected! 🥳🎉
```

**Review by kamilkisiela — APPROVED** [review]

File: packages/web/docs/src/pages/features/sso-oidc-provider.mdx [hunk 9]
```diff
@@ -55,3 +55,171 @@ Provider. Share this link with your users or add it to your SSO provider dashboa
...
+And now you'd have to fill in a couple of inputs to connect your OpenID Okta's provider as the
+following:
+
+![](/docs/pages/features/sso-oidc-provider/hive-connect-openid-provider-modal.png)
+
+- Token Endpoint = https://${yourOktaDomain}/oauth2/v1/token (ex:
+  https://trial-5517233.okta.com/oauth2/v1/token)
+- User Info Endpoint = https://${yourOktaDomain}/oauth2/v1/userinfo (ex:
+  https://trial-5517233.okta.com/oauth2/v1/userinfo)
+- Authorization Endpoint = https://${yourOktaDomain}/oauth2/v1/authorize (ex:
```
Comment (YassinEldeeb) [inline packages/web/docs/src/pages/features/sso-oidc-provider.mdx]:
```suggestion
- Token Endpoint = https://YOUR_OKTA_DOMAIN_HERE/oauth2/v1/token (ex:
  https://trial-5517233.okta.com/oauth2/v1/token)
- User Info Endpoint = https://YOUR_OKTA_DOMAIN_HERE//oauth2/v1/userinfo (ex:
  https://trial-5517233.okta.com/oauth2/v1/userinfo)
- Authorization Endpoint = https://YOUR_OKTA_DOMAIN_HERE//oauth2/v1/authorize (ex:
```

**Comment by YassinEldeeb** [comment]
🎉

**Pull request merged**